Sources/SKCore/TaskScheduler.swift

```diff
@@ -0,0 +1,459 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import CAtomics
+import Foundation
+import LSPLogging
+
+fileprivate var logger: Logger {
+  Logger(subsystem: subsystem + ".task-scheduler", category: LoggingScope.scope)
+}
+
+/// See comment on ``TaskDescriptionProtocol/dependencies(to:taskPriority:)``
+public enum TaskDependencyAction<TaskDescription: TaskDescriptionProtocol> {
+  case waitAndElevatePriorityOfDependency(TaskDescription)
+  case suspendDependency(TaskDescription)
+}
+
+public protocol TaskDescriptionProtocol: Identifiable, Sendable, CustomLogStringConvertible {
```

Comment:
I'd love for you to file an issue on github.com/apple/swift so we can enable making such a thing as TaskExecutor, rather than having to resort to trickery and unstructured tasks like this. I believe we could allow such things but today are missing pieces in the custom executors infra I think -- like querying a job for its details etc, like priority, maybe some hooks to know a job "finished" etc.

Reply:
Filed https://github.com/apple/swift/issues/73372. It’s a little vague, feel free to refine it with the things that you actually think are relevant and doable.

Reply:
That's good, thank you -- I'll fill in the bits, and we'll figure out what is actionable :)